docs/source/distributed.rst

```diff
@@ -395,6 +399,20 @@ of 16
 .. autofunction:: all_gather_multigpu
 
 
+third-party backends
+--------------------
+
+For references on how to develop a c10d third-party backend through C++ Extension,
+please refer to `Tutorials - Custom C++ and CUDA Extensions <https://pytorch.org/
+tutorials/advanced/cpp_extension.html>`_
+
+The backend is derived from `c10d.ProcessGroup` base class like what GLOO/NCCL/MPI made.
+The derived class should be named as `ProcessGroupXYZ` if the backend module name is `xyz`. 
+When user invokes :func:`torch.distributed.init_process_group` with a specified backend
+name such as `xyz`, the `torch.distributed` package attempts to import the module `xyz`
+automatically as well as it installed.
```

Comment:
Shall we add a link to the test case you added as an example? Alternatively, you may also link to how C10D is extended with CCL if the code is ready.

Reply:
done.